toss/trajectory_tools.py

```diff
@@ -34,32 +39,87 @@ def compute_trajectory(x: np.ndarray, args, func: Callable) -> Union[np.ndarray,
                 rtol (float): Relative error tolerance for integration.
                 atol (float): Absolute error tolerance for integration.
             problem: Parameters related to the problem:
-                start_time (float): Start time (in seconds) for the integration of trajectory.
-                final_time (float): Final time (in seconds) for the integration of trajectory.
+                start_time (int): Start time (in seconds) for the integration of trajectory.
+                final_time (int): Final time (in seconds) for the integration of trajectory.
                 initial_time_step (float): Size of initial time step (in seconds) for integration of trajectory.
                 radius_bounding_sphere (float): Radius of the bounding sphere representing risk zone for collisions with celestial body.
-                event (int): Event configuration (0 = no event, 1 = collision with body detection)
+                event (int): Event configuration (0 = no event, 1 = collision with body detection).
+                number_of_maneuvers (int): Number of possible maneuvers.
+            mesh:
+                vertices (np.ndarray): Array containing all points on mesh.
+                faces (np.ndarray): Array containing all triangles on the mesh.
+            state: Parameters provided by the state vector
+                time_of_maneuver (float): Time for adding impulsive maneuver [seconds].
+                delta_v (np.ndarray): Array containing the cartesian componants of the impulsive maneuver.
         func (Callable): A function handle for the state update equation required for integration.
 
     Returns:
         trajectory_info (np.ndarray): Numpy array containing information on position and velocity at every time step (columnwise).
         squared_altitudes (float): Sum of squared altitudes above origin for every position
         collision_penalty (bool): Penalty value given for the event of a collision with the celestial body.
     """
-    
-    # Compute trajectory by numerical integration
-    trajectory, trajectory_info = integrate(func, x, args)
+    # Convert osculating orbital elements to cartesian for integration
+    r, v = pk.par2ic(E=x[0:6], mu=args.body.mu)
+    x_cartesian = np.array(r+v)
+
+    # Setup time intervals with/without maneuvers
+    if args.problem.number_of_maneuvers == 0:
+        time_list = [args.problem.start_time, args.problem.final_time]
+    else:
+        # Add state variables related to the impulsive maneuver in args
+        time_list = np.empty((args.problem.number_of_maneuvers+2), dtype=int)
+        delta_v_array = np.empty((3, args.problem.number_of_maneuvers), dtype=np.float64)
+        m = 0
+        time_list[0] = args.problem.start_time
+        time_list[-1] = args.problem.final_time
+        for n in range(6, len(x)-3, 4):
+            time_list[m+1] = int(x[n])
+            delta_v_array[:,m] = [x[n+1], x[n+2], x[n+3]]
+            m += 1
```

Comment:
Looking at this, for readability it should probably iterate over m and then compute a variable n (what is n?) based on that? very hard to understand what is happening here

Reply:
Fixed.